aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -277,7 +278,27 @@ IntArrayRef NestedTensorImpl::sizes_custom() const {
   TORCH_CHECK(false, "Internal error: NestedTensorImpl doesn't support sizes. Please file an issue.");
 }
 c10::SymIntArrayRef NestedTensorImpl::sym_sizes_custom() const {
-  TORCH_CHECK(false, "Internal error: NestedTensorImpl doesn't support sizes. Please file an issue.");
+  // C++ NT should not support any ops that do metadata mutation
+  if (C10_UNLIKELY(!sym_sizes_.has_value())) {
+    if (nested_sizes_.dim() == 0) {
+      sym_sizes_ = std::vector<c10::SymInt>({c10::SymInt(0)});
+      return c10::SymIntArrayRef(*sym_sizes_);
+    }
+    int64_t out_len = nested_sizes_.size(1) + 1;
+    std::vector<c10::SymInt> sym_sizes(out_len);
+    for (const auto i : c10::irange(out_len)) {
+      if (auto mb_size = opt_size(i)) {
+        sym_sizes[i] = c10::SymInt(*mb_size);
+      } else {
+        auto vec = nested_sizes_.select(1, i - 1);
+        // See NOTE [ NestedTensorVariant ]
+        sym_sizes[i] = c10::SymInt(
+          c10::SymNode(c10::make_intrusive<c10::NestedIntSymNodeImpl>(-1, -1, std::move(vec), -1, c10::NestedTensorVariant::CPP)));
+      }
+    }
+    sym_sizes_ = std::move(sym_sizes);
```

Comment:
This doesn't look thread safe

Reply:
Why isn't sym_sizes_ just appropriately populated at construction time

Reply:
Hmm, we can probably do this.